### PR TITLE
Refactor convert

### PR DESCRIPTION
**Pull request by mbucknell**
Very similar to what I committed on Friday but, corrected the spread syntax to properly handle nested arrays (this caused the issue with a test breaking and the subsequent changes there). Also fixed the conversion of the celsius value if it is null.

### CODE REVIEW

File: assets/src/scripts/monitoring-location/iv-data-utils.js [hunk 1]
```diff
@@ -132,29 +142,38 @@ const createConvertedVariable = function(NWISVariable) {
...
+        offset: [...timeSeries.offset],
+        qualifier: [...timeSeries.qualifier],
+        qualityControlLevel: [...timeSeries.qualityControlLevel],
+        sample: [...timeSeries.sample],
+        source: [...timeSeries.source],
+        points: timeSeries.points.map((point) => {
+            return {
+                ...point,
+                qualifiers: [...point.qualifiers],
+                value: point.value ? convertCelsiusToFahrenheit(point.value).toFixed(2) : null
```
Comment (mbucknell) [inline assets/src/scripts/monitoring-location/iv-data-utils.js:158]: This is how I handled setting Fahrenheit to null when the celsius value is null.

Comment (abriggs-usgs) [reply assets/src/scripts/monitoring-location/iv-data-utils.js:158]: oh nice. That is a couple of lines less code than what I had done.

File: assets/src/scripts/monitoring-location/iv-data-utils.js [hunk 2]
```diff
@@ -109,16 +108,27 @@ const checkForMeasuredFahrenheitParameters = function(parameterCode, NWISVariabl
...
-    return  {
-        [NWISVariable.oid]: NWISVariable
+const getConvertedVariable = function(celsiusVariable) {
+    const calculatedFahrenheitVariable = {
+        ...celsiusVariable,
+        note: [...celsiusVariable.note],
+        options: [...celsiusVariable.options],
+        variableProperty: [...celsiusVariable.variableProperty],
+        variableName: celsiusVariable.variableName.replace('C', 'F (calculated)'),
+        variableDescription: celsiusVariable.variableDescription.replace('Celsius', 'Fahrenheit (calculated)'),
```
Comment (abriggs-usgs) [inline assets/src/scripts/monitoring-location/iv-data-utils.js:118]: Neat. Makes it pretty clear what is getting changed.

**Review by abriggs-usgs — APPROVED** [review]
Looks good.

**Pull request merged**